Tests/TuistDependenciesTests/Carthage/CarthageInteractorTests.swift

```diff
@@ -10,31 +10,99 @@ import XCTest
         carthageController = nil
-        carthageCommandGenerator = nil
-
+        carthage = nil
         subject = nil
 
         super.tearDown()
     }
 
+    func test_fetch() throws {
+        // Given
+        carthageController.canUseSystemCarthageStub = { true }
+
+        let rootPath = try TemporaryDirectory(removeTreeOnDeinit: true).path
+        let dependenciesDirectory = rootPath
+            .appending(component: Constants.DependenciesDirectory.name)
+
+        let platforms = Set<Platform>([.iOS, .watchOS, .macOS, .tvOS])
+        let options = Set<CarthageDependencies.Options>([])
+        let stubbedDependencies = CarthageDependencies(
+            [
+                .github(path: "Moya", requirement: .exact("1.1.1")),
+            ],
+            options: options
+        )
+
+        carthage.bootstrapStub = { [unowned self] parameters in
+            XCTAssertEqual(parameters.path, try self.temporaryPath())
+            XCTAssertEqual(parameters.platforms, platforms)
+            XCTAssertEqual(parameters.options, options)
+
+            // simulate output of Carthage bootstrap call
+            try [
+                "Cartfile.resolved",
+                "Carthage/Build/iOS/Moya.framework/Info.plist",
+                "Carthage/Build/iOS/ReactiveMoya.framework/Info.plist",
+                "Carthage/Build/iOS/RxMoya.framework/Info.plist",
+                "Carthage/Build/Mac/Moya.framework/Info.plist",
+                "Carthage/Build/Mac/ReactiveMoya.framework/Info.plist",
+                "Carthage/Build/Mac/RxMoya.framework/Info.plist",
+                "Carthage/Build/watchOS/Moya.framework/Info.plist",
+                "Carthage/Build/watchOS/ReactiveMoya.framework/Info.plist",
+                "Carthage/Build/watchOS/RxMoya.framework/Info.plist",
+                "Carthage/Build/tvOS/Moya.framework/Info.plist",
+                "Carthage/Build/tvOS/ReactiveMoya.framework/Info.plist",
+                "Carthage/Build/tvOS/RxMoya.framework/Info.plist",
+            ].forEach {
+                try fileHandler!.touch(parameters.path.appending(RelativePath($0)))
+            }
+        }
+
+        // When
+        try subject.fetch(
+            dependenciesDirectory: dependenciesDirectory,
+            dependencies: stubbedDependencies,
+            platforms: platforms
+        )
+
+        // Then
+        let expectedCartfileResolvedPath = dependenciesDirectory
+            .appending(component: Constants.DependenciesDirectory.lockfilesDirectoryName)
+            .appending(component: Constants.DependenciesDirectory.cartfileResolvedName)
+        let expectedCarthageDirectory = dependenciesDirectory
+            .appending(component: Constants.DependenciesDirectory.carthageDirectoryName)
+
+        XCTAssertTrue(fileHandler.exists(expectedCartfileResolvedPath))
+        XCTAssertTrue(fileHandler.exists(expectedCarthageDirectory.appending(components: "iOS", "Moya.framework", "Info.plist")))
```

Comment:
I think this calls for new `XCT` extension (I believe we have hit something similar earlier).
Something like:
```swift
XCTAssertFilesExist(_ files: AbsolutePath...) 
// or
XCTAssertFilesExist(_ files: [AbsolutePath])
```
so it can be used such as:
```swift
XCTAssertFilesExist(
   expectedCarthageDirectory.appending(components: "iOS", "ReactiveMoya.framework", "Info.plist")
)
```

in this case you could even make it simpler by reusing the array from above, mapping that to relative path and then just asserting on this new array.

I'd try to refrain from this large copy&pasting.